src/responder/common/responder_common.c

```diff
@@ -219,11 +219,11 @@ errno_t csv_string_to_uid_array(TALLOC_CTX *mem_ctx, const char *csv_string,
         goto done;
     }
 
-    if (allow_sss_loop) {
-        ret = unsetenv("_SSS_LOOPS");
-        if (ret != EOK) {
-            DEBUG(SSSDBG_OP_FAILURE, "Failed to unset _SSS_LOOPS, getpwnam "
-                                      "might not find sssd users.\n");
+    if (!set_sss_loops) {
```

Comment:
Please change this to `if (set_sss_loops)`, and flip the argument for calls to `csv_string_to_uid_array`.

For example:
in `src/responder/ifp/ifpsrv.c` below change from 'true' to 'false'

    ret = csv_string_to_uid_array(ifp_ctx->rctx, uid_str, true,
                                  &ifp_ctx->rctx->allowed_uids_count,
                                  &ifp_ctx->rctx->allowed_uids);

Reply:
Done.
I also renamed the argument `prevent_sss_loop` as it better describes its behavior.